check_modulemd.py

```diff
@@ -209,12 +210,24 @@ def test_component_availability(self):
         """
         Are all the components we reference in the packages section available?
         """
-        # verify that the specified ref (if any, defaults to master HEAD) is available in the
-        # specified repository (if any, defaults to Fedora dist-git).
-        self.log.warn("Not yet implemented")
+        # verify that the specified ref (if none, defaults to master HEAD) is available in the
+        # specified repository (if none, defaults to Fedora dist-git).
         for p in self.mmd.components.rpms.values():
-            self.log.warn(
-                "Need to check availability of component RPM %s" % p.name)
+            repository = "https://src.fedoraproject.org/cgit/rpms/%s.git" % p.name
+            if p.repository:
+                repository = p.repository
+            ref = "HEAD"
+            if p.ref:
+                ref = p.ref
+            patch_path = "/patch/?id=%s" % ref
+            url = repository + patch_path
+            resp = requests.head(url)
+            if not resp or resp.status_code != 200:
```

Comment:
To me raise_for_status()  is not clear when it raises exception, it seems to be only if status code are on range of 4XX and 5XX. 3XX range I'd assume we should treat as error.

I can check if status_code is not within 2XX range and report error.